docs/DevelopersGuide.adoc

```diff
@@ -47,359 +49,1141 @@ the Datomic pull syntax):
 
 ```
-(def parser (p/parser {}))
+(defresolver `person-age-resolver
+  {::pc/input #{:person/id}
+   ::pc/output [:person/age]}
+  (fn [{:keys [database] :as env} {:keys [person/id]}]
+    (let [person (my-database/get-person database id)]
+      {:person/age (:age person)})))
+
+(defresolver `person-first-name-resolver
+  {::pc/input #{:person/id}
+   ::pc/output [:person/first-name]}
+  (fn [{:keys [database] :as env} {:keys [person/id]}]
+    (let [person (my-database/get-person database id)]
+      {:person/first-name (:first-name person)})))
+
+...
 ```
 
-and every call to the parser needed an explicit reader: `(parser {::p/reader computed} [:hello])`
+The point is that a single-level query like `[:person/id :person/first-name :person/age]` can be satisfied and "folded together"
+by pathom over any number of resolvers.
 
-A vector of plugins can be added when the parser is created. The `p/env-plugin` automatically adds the given map of data to the parsing environment every time the resulting parser is called.
+This fact is the basis of parser (de)composition and extensibility.  It can also come in handy for performance
+refinements when there are computed attributes.
 
-Our example can be converted to:
+=== Derived/Computed Attributes
+
+There are times when you'd like to provide an attribute that is computed in some fashion.  You can, of course, simply
+compute it within the resolver along with other properties like so:
 
 ```
-(def parser (p/parser {::p/plugins [(p/env-plugin {::p/reader [computed]})]}))
+(defresolver `person-resolver
+  {::pc/input #{:person/id}
+   ::pc/output [:person/first-name :person/last-name :person/full-name :person/age]}
+  (fn [{:keys [database] :as env} {:keys [person/id]}]
+    (let [{:keys [age first-name last-name]} (my-database/get-person database id)]
+      {:person/age age
+       :person/first-name first-name
+       :person/last-name last-name
+       :person/full-name (str first-name " " last-name) ; COMPUTED
+       ... })))
 ```
 
-and now each call to the parser needs nothing in the supplied env: `(parser {} [:hello])`.
-
-Providing a default environment is such a common operation that we provide an easier way to set it up:
+but this means that you'll take the overhead of the computation when *any* query relate to person comes up.  You can
+instead spread such attributes out into other resolvers as we discussed previously, which will only be invoked if the
+query actually asks for those properties:
 
 ```
-(def parser (p/parser {::p/env {::p/reader [computed]}}))
+(defresolver `person-resolver
+  {::pc/input #{:person/id}
+   ::pc/output [:person/first-name :person/last-name :person/age]}
+  (fn [{:keys [database] :as env} {:keys [person/id]}]
+    (let [{:keys [age first-name last-name]} (my-database/get-person database id)]
+      {:person/age age
+       :person/first-name first-name
+       :person/last-name last-name })))
+
+(defresolver `person-name-resolver
+  {::pc/input #{:person/id}
```

Comment:
I like to change this example, here we introduced a duplication to the logic about getting the person from the database, we can instead depend directly on the fields to do the computation:

```clojure
(defresolver `person-name-resolver
  {::pc/input #{:person/first-name :person/last-name}
   ::pc/output [:person/full-name]}
  (fn [_ {:person/keys [first-name last-name]}]
    {:person/full-name (str first-name " " last-name)}))
```

This version is simpler, it doesn't complect the data fetching with the computation, and pathom can cache this more efficiently (if another attribute already called that fetch user resolver it will be cached, but doing the fetch again here would cause a second call to the database)

Reply:
I see, sure, that makes sense.